refactor(core): Improve `explicit-standalone-flag` migration.

JeanMeche opened the pull request:
Keep standalone:true

This commit also adds the support of handling standalone proparties assigned by variables (or via shorthand assignments).

AndrewKushnir commented on packages/core/schematics/test/explicit_standalone_flag_spec.ts:
```diff
@@ -171,4 +171,114 @@ describe('standalone:false migration', () => {
+    expect(content).not.toContain('standalone: true');
+    expect(content).toContain('foo');
+    expect(content).toMatch(
+      /selector:.*bar.*template:.*before standalone.*styles:.*after styles.*imports.*\}\)/,
```
I'm a bit confused on why do we need to retain comments that are "attached" to the `standalone` field, could you please provide a bit more info?

JeanMeche replied:
I was under the impression that leading comments were not in fact always attached to the  following declaration. 

On adev we have for example: 

https://github.com/angular/angular/blob/0ebb2be538301a9a05c9399dbf069de183068f7c/adev/src/content/examples/styleguide/src/02-07/app/heroes/hero.component.ts#L6-L12

Removing that doc actually broke the docs generation. We ended up having an un-even number of docregion/enddocregion.

AndrewKushnir replied:
Thanks for additional info. I think the use-case that we have is not very common and I believe it would be more confusing if we retain comment above the `standalone` field. My proposal is to drop that logic from the PR (i.e. keep removing a comment above the `standalone` field) and fix cases in docs manually. If there is way too many cases in our docs, we can also add some logic into the migration to recognize those special "docregion" constructs and only retain comments in those cases, but eventually remove it once FW repo is migrated.

JeanMeche replied:
We'll probably go with Andrew's suggestion for now and revisit it if during the beta phase we get several reports like yours.

JeanMeche committed:
refactor(migrations): Keep standalone:true

This commit also adds the support of handling standalone proparties assigned by variables (or via shorthand assignments).

JeanMeche commented:
For compatibility reasons, the migration will skip removing `standalone: true`, we'll re-enable it later (before the release of course).

AndrewKushnir commented:
[Presubmit](http://test/OCL:674144342:BASE:674144386:1726204522553:cbf414c1).

pkozlowski-opensource commented:
This PR was merged into the repository by commit 41c212d604810c128b3ac002e351159321e20119.

The changes were merged into the following branches: main
